compileopts/target.go

```diff
@@ -240,7 +254,14 @@ func defaultTarget(goos, goarch, triple string) (*TargetSpec, error) {
 		spec.Features = "+cx8,+fxsr,+mmx,+sse,+sse2,+x87"
 	case "arm":
 		spec.CPU = "generic"
-		spec.Features = "+armv7-a,+dsp,+fp64,+vfp2,+vfp2sp,+vfp3d16,+vfp3d16sp,-thumb-mode"
+		switch strings.Split(triple, "-")[0] {
+		case "armv5":
+			spec.Features = "+armv5t,+strict-align,-thumb-mode"
+		case "armv6":
+			spec.Features = "+armv6,+dsp,+fp64,+strict-align,+vfp2,+vfp2sp,-thumb-mode"
+		case "armv7":
+			spec.Features = "+armv7-a,+dsp,+fp64,+vfp2,+vfp2sp,+vfp3d16,+vfp3d16sp,-thumb-mode"
+		}
```

Comment:
Should we check for unrecognized options here?

Reply:
Do you mean an unrecognized architecture string (`armv5` etc)?
The only possible values are set in `LoadTarget`: `armv5`, `armv6`, and `armv7`. And the three possible GOARM values are again checked in builder_test.go to have the expected `spec.Features`.

Reply:
Oh, I guess I just missed the other check. Nevermind.